ci_jobs_trigger/libs/openshift_ci_re_trigger/openshift_ci_re_trigger.py

```diff
@@ -0,0 +1,184 @@
+        self.trigger_token = self.hook_data.get("trigger_token", self.config.get("trigger_token"))
+        self.trigger_url = self.hook_data.get("trigger_url", self.config.get("trigger_url"))
+        self.build_id = self.hook_data.get("build_id")
+        self.job_name = self.hook_data.get("job_name")
+        self.prow_job_id = self.hook_data.get("prow_job_id")
+        self.slack_webhook_url = self.config.get("slack_webhook_url")
+        self.verify_hook_data()
+
+        self.logger.info(
+            f"{self.log_prefix} Start processing flow for Job {self.job_name}|build {self.build_id}|prow {self.prow_job_id}"
+        )
+
+    def verify_hook_data(self):
+        if not self.trigger_url:
+            self.logger.error(f"{self.log_prefix} openshift ci trigger url is mandatory.")
+
+        if not self.trigger_token:
+            self.logger.error(f"{self.log_prefix} openshift ci token is mandatory.")
+
+        if not self.job_name:
+            self.logger.error(f"{self.log_prefix} openshift ci job name is mandatory.")
+
+        if not self.build_id:
+            self.logger.error(f"{self.log_prefix} openshift ci build id is mandatory.")
+
+        if not self.prow_job_id:
+            self.logger.error(f"{self.log_prefix} openshift ci prow job id is mandatory.")
+
+        if not all((self.trigger_token, self.job_name, self.build_id, self.prow_job_id, self.trigger_url)):
+            raise ValueError(f"{self.log_prefix} Missing parameters")
+
+    def execute_trigger(self, job_db_path=None):
+        slack_msg = f"""
+Job: {self.job_name}
+Build ID: {self.build_id}
+Prow ID: {self.prow_job_id}
+"""
+        with DB(job_db_path=job_db_path) as database:
+            if database.check_prow_job_id_in_db(job_name=self.job_name, prow_job_id=self.prow_job_id):
+                self.logger.warning(f"{self.log_prefix} Job was already auto-triggered. Exiting.")
+                if self.slack_webhook_url:
+                    send_slack_message(
+                        message=f"{self.log_prefix}{slack_msg}already auto-triggered, will not re-trigger",
+                        webhook_url=self.slack_webhook_url,
+                        logger=self.logger,
+                    )
+                    return True
+
+                return False
+
+        if not self.wait_for_job_completed():
+            if self.slack_webhook_url:
+                send_slack_message(
+                    message=f"{self.log_prefix}{slack_msg}Timeout waiting for job to complete, not re-triggering",
+                    webhook_url=self.slack_webhook_url,
+                    logger=self.logger,
+                )
+
+            raise requests.exceptions.RequestException()
+
+        tests_dict = self.get_testsuites_testcase_from_junit_operator(
+            junit_xml=self.get_tests_from_junit_operator_by_build_id()
+        )
+        if self.is_build_failed_on_setup(tests_dict=tests_dict):
+            prow_job_id = self.trigger_job()
```

Comment:
we need to rename either `trigger_job()` or `self.trigger_job()`, preferred to rename the self one